scheduling/storage_provider.go

```diff
@@ -41,9 +43,18 @@ func (m *MemoryStorageProvider) Get(key string, prefix bool) ([]string, error) {
 	}
 
 	var result []string
-	for k, v := range m.storage {
+	var keys []string
+	for k, _ := range m.storage {
 		if strings.HasPrefix(k, key) {
-			result = append(result, v)
+			keys = append(keys, k)
+		}
+	}
+
+	sort.Strings(keys)
+
+	for _, k := range keys {
+		if strings.HasPrefix(k, key) {
```

Comment:
I may be missing context, but do we need this second `.hasPrefix` check if keys is already made of items with the prefix?